pre/remap_restart/remap_upper.py

```diff
@@ -131,35 +136,37 @@ def remap(self):
        NPE = 5400; nwrit= 6
 
      QOS = "#SBATCH --qos="+config['slurm']['qos']
-     if NPE > 532: QOS = "###" + QOS
-     CONSTR = "#SBATCH --constraint=" + config['slurm']['constraint']
+     TIME ="#SBATCH --time=1:00:00"
+     if NPE > 532: 
+       assert config['slurm']['qos'] != 'debug', "qos should be allnccs"
+       TIME = "#SBATCH --time=12:00:00"
+
+     PARTITION = "#SBATCH --partition=" + config['slurm']['partition']
 
      log_name = out_dir+'/remap_upper_log'
 
      # We need to create an input.nml file which is different if we are running stretched grid
      # First, let's define a boolean for whether we are running stretched grid
      # If we are running with imout of 270, 540, 1080, 1536 or 2160, then we are running stretched grid
-     if imout in [270, 540, 1080, 2160]:
-        stretched_grid = True
-        target_lat = 39.5
-        target_lon = -98.35
-        stretch_fac = 2.5
-     elif imout in [1536]:
-        stretched_grid = True
-        target_lat = 39.5
-        target_lon = -98.35
-        stretch_fac = 3.0
-     else:
-        stretched_grid = False
-
      # If we are running stretched grid, we need to pass in the target lat, lon, and stretch factor
      # to interp_restarts.x. Per the code we use:
      #  -stretched_grid target_lon target_lat stretch_fac
      # If we are not running stretched grid, we should pass in a blank string
-     if stretched_grid:
+     stretch = config['output']['shared']['stretch']
+     stretch_str = ""
+     if stretch:
+        target_lat  = 0.0
+        target_lon  = 0.0
+        stretch_fac = 1.0
+        if stretch == 'SG001':
```

Comment:
1. I don't think we should set these defaults and then leave them in place if the particular string for SGxxx is not in list of "if" statements.  If for some reason (e.g., incomplete code update), we arrive here with stretch='SG003', then the parameters will be the defaults, right?  But the defaults are by definition the not-stretched CS grid.  We need to make sure that the code exits cleanly if "stretch" is unknown.
2. More broadly, I think we need to define the SGxxx table somewhere centrally, say in remap_utils.py.  At the moment, there are two places where SG001 and SG002 are defined, not incl. the comments.

Reply:
Mostly addressed by the two most recent commits.  Could still use more cleanup.